e2etest/bgp_test.go

```diff
@@ -93,32 +93,42 @@ var _ = ginkgo.Describe("BGP", func() {
 			},
 		},
 	}
-	ginkgo.BeforeEach(func() {
-		var err error
-		if containersNetwork == "host" {
-			if net.ParseIP(hostIPv4) == nil {
-				framework.Fail("Invalid hostIPv4")
-			}
-			if net.ParseIP(hostIPv6) == nil {
-				framework.Fail("Invalid hostIPv6")
+
+	ginkgo.AfterEach(func() {
+		if ginkgo.CurrentGinkgoTestDescription().Failed {
+			for _, c := range frrContainers {
+				dump, err := frr.RawDump(c)
+				framework.Logf("External frr dump for %s %s %v", c.Name, dump, err)
 			}
+			DescribeSvc(f.Namespace.Name)
 		}
-		cs = f.ClientSet
-		frrContainers, err = createFRRContainers(containersConf)
-		framework.ExpectNoError(err)
 	})
 
 	ginkgo.AfterEach(func() {
+		ginkgo.By("Clearing the previous configuration")
 		// Clean previous configuration.
 		err := updateConfigMap(cs, configFile{})
 		framework.ExpectNoError(err)
 
 		err = stopFRRContainers(frrContainers)
 		framework.ExpectNoError(err)
+	})
 
-		if ginkgo.CurrentGinkgoTestDescription().Failed {
-			DescribeSvc(f.Namespace.Name)
+	f = framework.NewDefaultFramework("bgp")
+
+	ginkgo.BeforeEach(func() {
+		var err error
```

Comment:
u don't need var u can initialize down when u use it ?

Reply:
I can't because I need to assign the external `frrContainer` variable. If I do
```
		frrContainers, err = createFRRContainers(containersConf)
```
it will create an unused local version of `frrContainers`